examples/notebooks/get-started/try-apache-beam-yaml.ipynb

```diff
@@ -0,0 +1,629 @@
+    "      config:\n",
+    "        path: data/people.csv\n",
+    "    - type: MapToFields\n",
+    "      config:\n",
+    "        language: python\n",
+    "        append: true\n",
+    "        fields:\n",
+    "          random_number:\n",
+    "            callable: |\n",
+    "              import random\n",
+    "              def my_mapping(row):\n",
+    "                return random.randrange(row.age)\n",
+    "    - type: LogForTesting\n",
+    "'''\n",
+    "save_to_file(pipeline, 'pipelines/pipeline-map-04.yaml')\n",
+    "! python -m apache_beam.yaml.main --pipeline_spec_file=pipelines/pipeline-map-04.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Beam YAML has EXPERIMENTAL ability to do aggregations to group and combine values across records. The is accomplished via the `Combine` transform type. Currently `Combine` needs to be in the `yaml_experimental_features` option (see the bottom of the pipeline) to use this transform.\n",
+    "\n",
+    "In this example we'll aggregate our records based on the `is_adult` classification. We'll calculate an average age for each of the groups."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "pipeline = '''\n",
+    "pipeline:\n",
+    "  type: chain\n",
+    "  transforms:\n",
+    "    - type: ReadFromCsv\n",
+    "      config:\n",
+    "        path: data/people.csv\n",
+    "    - type: MapToFields\n",
+    "      config:\n",
+    "        language: python\n",
+    "        append: true\n",
+    "        fields:K\n",
+    "          is_adult: \"age >= 18\"\n",
+    "    - type: Combine\n",
+    "      config:\n",
+    "        group_by: is_adult\n",
+    "        combine:\n",
+    "          total:\n",
+    "            value: age\n",
+    "            fn:\n",
+    "              type: mean\n",
+    "    - type: LogForTesting\n",
+    "options:\n",
+    "  yaml_experimental_features: Combine\n",
+    "'''\n",
+    "save_to_file(pipeline, 'pipelines/pipeline-combine-01.yaml')\n",
+    "! python -m apache_beam.yaml.main --pipeline_spec_file=pipelines/pipeline-combine-01.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If all was executed correctly, you should see the following lines at the bottom of the output log:\n",
+    "```\n",
+    "INFO:root:Result(is_adult=True, total=40.674418604651166)\n",
+    "INFO:root:Result(is_adult=False, total=11.714285714285714\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All the previous pipelines were linear - output of one transform was an input to the next transform. In YAML we can also create nonlinear pipelines. To do this, we need to specify `type: composite` (or omit this line at all, this is the default mode) and specify the `name` and `input` in each of the following transforms.\n",
```

Comment:
Nit: I would point out the fact the last several examples used `chain`

All the previous pipelines were linear - output of one transform was an input to the next transform. This is also known as a `chain` pipeline. This is designated in the top-level pipeline configuration, for example:
```
pipeline:
  type: chain
  transforms:
    ...
``` 
In YAML we can also create nonlinear pipelines. To do this, we need to specify `type: composite` (or omit this line at all, this is the default mode) and specify the `name` and `input` in each of the following transforms.

Reply:
good suggestion, done